internal/pkg/cli/svc_deploy.go

```diff
@@ -582,16 +586,35 @@ func (e *errFeatureIncompatibleWithEnvironment) RecommendActions() string {
 
 }
 
+type errNonEmptyDiff struct{}
+
+func (e *errNonEmptyDiff) Error() string {
+	return "diff detected "
+}
+
+// ExitCode returns 1 for a non-empty diff.
+func (e *errNonEmptyDiff) ExitCode() int {
+	return 1
+}
+
 func diff(differ templateDiffer, tmpl string, writer io.Writer) error {
 	out, err := differ.DeployDiff(tmpl)
 	if err != nil {
 		return err
 	}
+	var nonDiff bool
 	if out == "" {
+		nonDiff = true
 		out = "No changes.\n"
 	}
 	_, err = writer.Write([]byte(out))
-	return err
+	if err != nil {
+		return err
+	}
+	if nonDiff {
+		return nil
+	}
+	return &errNonEmptyDiff{}
```

Comment:
non empty diff confuses me haha...could we change the wording/logic to `noDiff`?

Reply:
```
out, err := differ.DeployDiff(tmpl)
	if err != nil {
		return err
	}
	if out != "" {
		if _, err := writer.Write([]byte(out)); err != nil {
			return err
		}
		return &errHasDiff{}
	}
	if _, err := writer.Write([]byte("No changes.\n")); err != nil {
		return err
	}
	return nil
```

Better?